src/value.rs

```diff
@@ -216,6 +216,11 @@ impl ValueCommitTrapdoor {
     pub(crate) fn inner(&self) -> pallas::Scalar {
         self.0
     }
+
+    /// Constructs `ValueCommitTrapdoor` from the byte representation of a scalar.
+    pub fn from_bytes(bytes: [u8; 32]) -> CtOption<Self> {
```

Comment:
I'm not even sure why `ValueCommitTrapdoor` was ever a public type. AFAICT it is not used anywhere in our public API; there was no way to construct one publicly, and nothing consumes it.

This API will enable crate users to construct `ValueCommitTrapdoor` from bytes, after which the (inherently-exposed) `Add` and `Sum` impls will become usable. But what is the purpose of being able to publicly construct a `ValueCommitTrapdoor`?

Reply:
Ah, I'm guessing this should be reviewed in combination with #351 which exposes an API that consumes `ValueCommitTrapdoor`. That PR provides the context that this PR lacks: that the caller is deriving a `ValueCommitment` themselves.

This API needs significantly more documentation, because it is a relatively low-level API that has the potential to result in a complete loss of privacy (because it pushes the responsibility for correct generation of the trapdoor onto the caller). The documentation should describe the contexts in which this API should be used, and how to correctly and safely generate the bytes to be used as input.

Reply:
Why not remove those `Add` and `Sum` instances?